client/servicetunnel/client.go

```diff
@@ -0,0 +1,144 @@
+	DeletePolicy(tunID string, policyID string) (err error)
+}
+
+func (a *ServiceTunnel) Get(id string) (spec ServiceTunnelInfo, err error) {
+	resp, err := a.restClient.Read(apiVersion, component, id, "")
+	if err != nil {
+		return
+	}
+	spec, err = specFromResponse(resp)
+	return
+}
+
+func (a *ServiceTunnel) Create(spec Info) (created ServiceTunnelInfo, err error) {
+	body, err := json.Marshal(Info{
+		Kind:       "BanyanAccessTier",
+		APIVersion: "rbac.banyanops.com/v1",
+		Type:       "attribute-based",
+		Metadata: Metadata{
+			Name:         spec.Metadata.Name,
+			FriendlyName: spec.Metadata.FriendlyName,
+			Description:  spec.Metadata.Description,
+		},
+		Spec: spec.Spec,
+	})
+	resp, err := a.restClient.Create(apiVersion, component, body, "")
+	if err != nil {
+		return
+	}
+	created, err = specFromResponse(resp)
+	return
+}
+
+func (a *ServiceTunnel) Update(id string, spec Info) (updated ServiceTunnelInfo, err error) {
+	body, err := json.Marshal(Info{
+		Kind:       "BanyanAccessTier",
+		APIVersion: "rbac.banyanops.com/v1",
+		Type:       "attribute-based",
+		Metadata: Metadata{
+			Name:         spec.Metadata.Name,
+			FriendlyName: spec.Metadata.FriendlyName,
+			Description:  spec.Metadata.Description,
+		},
+		Spec: spec.Spec,
+	})
+	resp, err := a.restClient.Update(apiVersion, component, id, body, "")
+	if err != nil {
+		return
+	}
+	updated, err = specFromResponse(resp)
+	return
+}
+
+func (a *ServiceTunnel) Delete(id string) (err error) {
+	err = a.restClient.Delete(apiVersion, component, id, "")
+	return
+}
+
+func (a *ServiceTunnel) GetPolicy(id string) (policy PolicyAttachmentInfo, err error) {
+	path := fmt.Sprintf("%s/%s/%s/security_policy", apiVersion, component, id)
+	resp, err := a.restClient.Read(apiVersion, component, id, path)
+	if err != nil {
+		return
+	}
+	err = json.Unmarshal(resp, &policy)
+	return
+}
+
+func (a *ServiceTunnel) AttachPolicy(id string, post PolicyAttachmentPost) (created PolicyAttachmentInfo, err error) {
+	if id == "" {
+		err = fmt.Errorf("need service tunnel id to attach a policy")
+	}
+	body, err := json.Marshal(post)
+	path := fmt.Sprintf("%s/%s/%s/security_policy", apiVersion, component, id)
+	resp, err := a.restClient.Create(apiVersion, component, body, path)
+	if err != nil {
+		return
+	}
+	var j PolicyResponse
+	err = json.Unmarshal(resp, &j)
+	created = j.Data
+	return
+}
+
+func (a *ServiceTunnel) DeletePolicy(tunID string, policyID string) (err error) {
+	path := fmt.Sprintf("%s/%s/%s/security_policy/%s", apiVersion, component, tunID, policyID)
+	err = a.restClient.Delete(apiVersion, component, tunID, path)
+	return
+}
+
+func specFromResponse(respData []byte) (created ServiceTunnelInfo, err error) {
+	var j Response
+	err = json.Unmarshal(respData, &j)
+	if err != nil {
+		return
+	}
+	spec := j.Data
+	var jSpec Info
+	specString := html.UnescapeString(spec.Spec)
+	err = json.Unmarshal([]byte(specString), &jSpec)
+	created = ServiceTunnelInfo{
```

Comment:
I guess the `err` is returned below but does it make sense to populate `created` if we have a non-`nil` error?